public/app/features/alerting/unified/api/onCallApi.ts

```diff
@@ -0,0 +1,27 @@
+import { alertingApi } from './alertingApi';
+export interface OnCallIntegration {
+  [key: string]: string;
```

Comment:
what's the reason it's `[key: string]: string` instead of a defined set of properties? is it not known yet or too large?

Reply:
Good question, as this type depends on an external onCall api, we can not guarantee that the list of properties will not change. The only property we can relay on is the `integration_url`.

Reply:
If we can't rely on any other property it might be safer to simply omit `[key: string]: string` altogether since it will tell the typechecker than any additional property is fine (and is also typed as a string, JSON payloads can also contain booleans, numbers, `null`, etc).

I think it's best to either type the other properties (we assume the OnCall API has strict backwards compat guarantees) or remove any and all assumptions and only keep `integration_url`.

Reply:
true, it's better not to mention the unknown properties.💯